src/classes/BGE_BatchGiftEntryTab_CTRL.cls

```diff
@@ -0,0 +1,180 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group BGE
+* @group-content
+* @description BGE Tab Component Controller.
+*/
+public with sharing class BGE_BatchGiftEntryTab_CTRL {
+
+    /*******************************************************************************************************
+    * @description Gets Data Import Batch data to populate datatable on BGE Batch Gift Entry Tab
+    * @return List<DataImportBatch__c> list of Data Import Batches
+    */
+    @AuraEnabled
+    public static String getTabModel() {
+        if (UTIL_Permissions.canRead(UTIL_Namespace.StrTokenNSPrefix('DataImportBatch__c'), false)) {
+            TabModel model = new TabModel();
+            model.columns = getTabColumns();
+            model.batches = getInitialBatches();
+            model.totalNumberOfRows = [SELECT Count() FROM DataImportBatch__c WHERE GiftBatch__c = TRUE];
+            return JSON.serialize(model);
+        } else {
+            throw new AuraHandledException(Label.bgeFLSError);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Gets columns to create datatable on BGE Batch Gift Entry Tab
+    * @return List<Column> list of columns for datatable
+    */
+    public static List<Column> getTabColumns() {
+
+        Set<String> disallowedFields = new Set<String>{
+                'Name',
+                UTIL_Namespace.StrTokenNSPrefix('Active_Fields__c')
+        };
+
+        List<String> fieldNames = getDataTableColumnFieldNames();
+        List<Column> columns = new List<Column>();
+
+        Column batchLinkCol = new Column();
+        batchLinkCol.label = UTIL_Describe.getFieldLabel(UTIL_Namespace.StrTokenNSPrefix('DataImportBatch__c'), 'Name');
+        batchLinkCol.fieldName = 'batchLink';
+        batchLinkCol.sortable = true;
+        batchLinkCol.type = 'url';
+        columns.add(batchLinkCol);
+
+        for (String fieldName : fieldNames) {
+            if (!disallowedFields.contains(fieldName)) {
+                DescribeFieldResult dfr = UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImportBatch__c'), fieldName);
+                String dataType = dfr.getType().name().toLowerCase();
+                Column col = new Column();
+                col.fieldName = fieldName;
+                col.label = dfr.label;
+                col.type = UTIL_LightningComponentSupport.getLightningDataType(dfr.getType().name());
```

Comment:
What happens if there's an unsupported data type in the fieldset?

Reply:
the raw Apex data type is returned if it isn't one of the known lightning:datatable weird cases: number, date, datetime.